### PR TITLE
feat(nft-market): Infinite loading for squad on sale

### PR DESCRIPTION
**Pull request by ChefHutch**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

File: src/views/Nft/market/Collection/OnSale.tsx [hunk 1]
```diff
@@ -35,62 +40,87 @@ const OnSale = () => {
...
-            })}
-          </Grid>
-        )}
+          <Flex mt="60px" mb="12px" justifyContent="center">
+            {isFetching ? (
+              <Spinner />
+            ) : (
+              <Button onClick={handleLoadMore} scale="sm">
+                {t('Load more')}
+              </Button>
```
Comment (0xCorgi) [inline src/views/Nft/market/Collection/OnSale.tsx:120]: Do we have a way to hide it if there's nothing more to fetch?

**Comment by memoyil** [comment]
Imho intersection observer will give better user experience than clicking button

File: src/views/Nft/market/Collection/OnSale.tsx [hunk 2]
```diff
@@ -35,62 +40,87 @@ const OnSale = () => {
           marketData: marketData[i],
         }
       })
-      setNfts(responsesWithMarketData)
+      setIsFetching(false)
+      setNfts((prevState) => {
+        const combinedNfts = [...prevState, ...responsesWithMarketData]
+        const filteredForDuplicates = uniqBy(combinedNfts, 'tokenId')
+        return filteredForDuplicates
```
Comment (0xCorgi) [inline src/views/Nft/market/Collection/OnSale.tsx:47]:
```suggestion
        return uniqBy(combinedNfts, 'tokenId')
```

**Review by 0xCorgi — APPROVED** [review]
Nice feature!

**Comment by ChefHutch** [comment]
Went ahead and changed the stuff Corgi suggested as well as adding the 'recently listed' sort on top of these commits so closing for now - https://github.com/pancakeswap/pancake-frontend/pull/2485

**Pull request closed without merging**